File: week6/sentimental-readability/readability.py
from cs50 import get_string
from string import ascii_lowercase as alphabet


def count_letters(text):
    count = 0
    text = text.lower()
    for i in alphabet:
        count += text.count(i)
    return count


def count_words(text):
    count = len(text.split())
    return count


def count_sentence(text):
    count = 0
    stop_points = [".", "!", "?"]
    for i in stop_points:
        count += text.count(i)
    return count


def calculation(text):
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentence(text)
    l_value = round(letters / words * 100, 4)
    s_value = round(sentences / words * 100, 3)
    index = 0.0588 * l_value - 0.296 * s_value - 15.8
    return index


def main():
    text = get_string("Text:")
    index = round(calculation(text))
    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")


main()

File: week2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");
    printf("%s\n", text);
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);
    float L = (float) letters / (float) words * 100.0;
    float S = (float) sentences / (float) words * 100.0;
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

int count_letters(string text)
{
    int countL = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            countL++;
        }
    }
    return countL;
}

int count_words(string text)
{
    int countW = 1;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == ' ' && (text[i + 1] >= 33 && text[i + 1] <= 126))
            // trial with int for including also symbols, idk if it is worth it.
        {
            countW++;
        }
    }
    return countW;
}

int count_sentences(string text)
{
    int countS = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if ((text[i] == '.' || text[i] == '!' || text[i] == '?')\
            && ((text[i - 1] >= 'a' && text[i - 1] <= 'z') || (text[i - 1] >= 'A' && text[i - 1] <= 'Z')))
        {
            countS++;
        }
    }
    return countS;
}